src/main/scala/de/retest/guistatemachine/api/GuiStateMachine.scala

```diff
@@ -31,10 +44,9 @@ trait GuiStateMachine {
     * @param to The state which the execution leads to.
     * @return The number of times the action has been executed.
     */
-  def executeAction(from: State, a: ActionIdentifier, to: State): Int = from.addTransition(a, to)
-  def executeAction(from: State, a: Action, to: State): Int = executeAction(from, new ActionIdentifier(a), to)
-  def executeAction(fromSutState: SutState, a: Action, toSutState: SutState): Int =
-    executeAction(getState(fromSutState), a, getState(toSutState))
+  def executeAction(from: State, a: ActionIdentifier, to: State, isUnexploredActionType: Boolean): Int = from.addTransition(a, to, isUnexploredActionType)
+  def executeAction(from: State, a: Action, to: State, isUnexploredActionType: Boolean): Int =
```

Comment:
Scaladoc seems to be missing for `isUnexploredActionType`. But do we really want this to be part of the public API? Shouldn't the SUT model internally decide if an action has been executed before or not?

Reply:
We can only decide it internally if we know the ActionType which is part surili-commons. Then I could generate the ActionType internally and compare it against a list of explored ActionTypes.

Reply:
If it helps: You could extract `ActionType` from `feature/action-selection-strategies` in Bitbucket and make separate PR. Will approve and deploy, so you can use it here?

Reply:
Deployed a new snapshot with `ActionType`.